### PR TITLE
[4.1 ScheduledTasks] test task to run task and ACL PERMISSION also updated

### PR DESCRIPTION
**Pull request by niharikamahajan02**
Pull Request for Issue #36453 (Part 2) .

### Summary of Changes
The ACL Permission also renaming and column header test task renamed to run task.

### CODE REVIEW

File: administrator/language/en-GB/com_scheduler.ini [hunk 1]
```diff
@@ -131,7 +131,7 @@ COM_SCHEDULER_TEST_RUN_STATUS_STARTED="Status: Started"
 COM_SCHEDULER_TEST_RUN_STATUS_TERMINATED="Status: Terminated"
 COM_SCHEDULER_TEST_RUN_TASK="Task: \"%s\""
 COM_SCHEDULER_TEST_RUN_TITLE="Test task (ID: %d)"
-COM_SCHEDULER_TEST_TASK="Test Task"
+COM_SCHEDULER_TASK_RUN="Run task"
```
Comment (Quy) [inline administrator/language/en-GB/com_scheduler.ini:134]: Please sort key in alpha order.

File: administrator/language/en-GB/com_scheduler.ini [hunk 2]
```diff
@@ -99,7 +99,7 @@ COM_SCHEDULER_OPTION_EXECUTION_MANUAL_LABEL="Manual Execution"
 COM_SCHEDULER_OPTION_ORPHANED_HIDE="Hide Orphaned"
 COM_SCHEDULER_OPTION_ORPHANED_ONLY="Only Orphaned"
 COM_SCHEDULER_OPTION_ORPHANED_SHOW="Show Orphaned"
-COM_SCHEDULER_PERMISSION_TESTRUN="Test task"
+COM_SCHEDULER_PERMISSION_TESTRUN="Run task"
```
Comment (Kostelano) [inline administrator/language/en-GB/com_scheduler.ini]: It makes sense to change the constant here by analogy with the change below. "Test" remove / replace

Comment (niharikamahajan02) [reply administrator/language/en-GB/com_scheduler.ini]: Did the required changes in PR https://github.com/joomla/joomla-cms/pull/36664

File: administrator/language/en-GB/com_scheduler.ini (hunk 1 above)
Comment (niharikamahajan02) [reply administrator/language/en-GB/com_scheduler.ini:134]: Did the required changes in PR https://github.com/joomla/joomla-cms/pull/36664

**Comment by tecpromotion** [comment]
Perhaps a clean renaming of all strings of TEST_RUN and TESTRUN occurrences to RUN_TASK will do the trick. 

e.g.
COM_SCHEDULER_PERMISSION_RUN_TASK
COM_SCHEDULER_RUN_TASK_STATUS_TERMINATED

That also reads much more clearly.

What do you think @brianteeman ?

**Comment by niharikamahajan02** [comment]
> Perhaps a clean renaming of all strings of TEST_RUN and TESTRUN occurrences to RUN_TASK will do the trick.
> 
> e.g. COM_SCHEDULER_PERMISSION_RUN_TASK COM_SCHEDULER_RUN_TASK_STATUS_TERMINATED
> 
> That also reads much more clearly.
> 
> What do you think @brianteeman ?

For replacing all strings of TEST_RUN to RUN_TASK , it's needed to replace respective constants also in model-content class.
image

image

as shown in screenshot 2. But I am unable to find where this content is written , so will you please inform.

**Comment by niharikamahajan02** [comment]
> Perhaps a clean renaming of all strings of TEST_RUN and TESTRUN occurrences to RUN_TASK will do the trick.
> 
> e.g. COM_SCHEDULER_PERMISSION_RUN_TASK COM_SCHEDULER_RUN_TASK_STATUS_TERMINATED
> 
> That also reads much more clearly.
> 
> What do you think @brianteeman ?

For replacing all strings of TEST_RUN to RUN_TASK , it's needed to replace respective constants also in modal-content class.

image

But I am unable to find where this content is written , so will you please inform .
@tecpromotion sir plz inform.

**Comment by tecpromotion** [comment]
The easiest way is to search for ALL files in the repository or your branch.
I use the full text search in my development environment (PhpStorm).

For example, if you search for COM_SCHEDULER_PERMISSION_TESTRUN across all files, you will find 3 places.

https://github.com/joomla/joomla-cms/blob/989c71976e97975a11387715bc147beb54e86b65/administrator/language/en-GB/com_scheduler.ini#L102
https://github.com/joomla/joomla-cms/blob/989c71976e97975a11387715bc147beb54e86b65/administrator/components/com_scheduler/access.xml#L11
https://github.com/joomla/joomla-cms/blob/989c71976e97975a11387715bc147beb54e86b65/administrator/components/com_scheduler/access.xml#L17

This must then all be adapted and tested.
Of course, the same applies to all other constants and their values.

Does this help you @niharikamahajan02 to create the further changes for this PR?

**Comment by niharikamahajan02** [comment]
Okay! thank you @tecpromotion sir.
Yes, I will be able to make further changes for this PR now.

**Comment by tecpromotion** [comment]
> Okay! thank you @tecpromotion sir. Yes, I will be able to make further changes for this PR now.

Perfect. Please close PR #36664 
I am looking forward to testing this PR. It makes sense.

**Comment by niharikamahajan02** [comment]
Even though after changing constants , the heading is not changing as shown below. Is there any specific problem here . 
image

**Comment by brianteeman** [comment]
Well among other things you have committed stashed changes from your github repo

image

**Comment by niharikamahajan02** [comment]
Yes because the changes I applied were not getting commited unless I didn't stashed them

**Comment by brianteeman** [comment]
Well thats wrong. Your branch was probably not upto datae

**Comment by niharikamahajan02** [comment]
Yup okk ,will be making another commit

**Comment by niharikamahajan02** [comment]
plz ignore this commit

**Comment by niharikamahajan02** [comment]
I think I will have to make a new PR as there is something getting wrong again and again.

**Comment by niharikamahajan02** [comment]
Sir plz view PR #36698 once . Some checks are failing .

**Pull request closed without merging**